client/src/fake_api_client.ts

```diff
@@ -107,34 +107,45 @@ export const fakeApiClient = {
     const filtered = bits.filter((bit) => {
       if (filters.mainCharacters.size > 0) {
         if (!bit.mainChars) return false;
-        for (const char of Array.from(filters.mainCharacters)) {
-          if (!bit.mainChars || bit.mainChars.indexOf(char.id) === -1)
-            return false;
+        if (
+          Array.from(filters.mainCharacters).filter(
```

Comment:
use `.find()` instead of `.filter().length` (throughout)

Reply:
Done.